src/modules/launcher/PowerLauncher/Storage/UserSelectedRecord.cs

```diff
@@ -12,21 +12,24 @@ public class UserSelectedRecord
 
         public void Add(Result result)
         {
-            var key = result.ToString();
-            if (records.TryGetValue(key, out int value))
+            if(result != null)
```

Comment:
@somil55 In what cases are we expecting result to be null? If we don't know this should probably be a nullargexception, otherwise we should add a comment on what cases it's expected to be null and why we guard against it.

Reply:
@ryanbodrug-microsoft All calls to `Add` are preceded by null check. So in our codebase, it would ever be null. I will update this to throw nullargexception.